metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -209,17 +380,423 @@ def _get_column_info(  # noqa: C901
+    try:
+        
+        # Query for CLUSTER TYPE
+        cluster_type_qry = sql.text(dedent("""
+            SELECT 
+            CASE COUNT(*) 
+                WHEN 0 
+                THEN 'Enterprise' 
+                ELSE 'Eon' 
+            END AS database_mode 
+            FROM v_catalog.shards
+        """ ))
+
+        communal_storage_path = sql.text(dedent("""
+            SELECT location_path from storage_locations 
+                WHERE sharing_type = 'COMMUNAL'
+        """ ))
+
+        cluster_type = ""
+        communical_path = ""
+        communal_data_res = connection.execute(communal_storage_path)
+        cluster_type_res = connection.execute(cluster_type_qry)
+        for each in cluster_type_res:
+            cluster_type = each.database_mode
+            if cluster_type.lower() == 'eon':
+                for each in communal_data_res:
+                    communical_path += str(each.location_path) + " | "
+
+        SUBCLUSTER_SIZE = sql.text(dedent("""
+                        SELECT subclusters.subcluster_name , CAST(sum(disk_space_used_mb // 1024) as varchar(10)) as subclustersize from subclusters  
+                        inner join disk_storage using (node_name) 
+                        group by subclusters.subcluster_name
+                         """))
+        
+        subclusters = " "
+        for data in connection.execute(SUBCLUSTER_SIZE):
+            subclusters += f"{data['subcluster_name']} -- {data['subclustersize']} GB |  "
+        
+        cluster__size = sql.text(dedent("""
+            select ROUND(SUM(disk_space_used_mb) //1024 ) as cluster_size
+            from disk_storage
+        """ ))
+        cluster_size = ""
+        for each in connection.execute(cluster__size):
+            cluster_size = str(each.cluster_size) + " GB"
+
+        
+        return {"cluster_type": cluster_type, "cluster_size": cluster_size, 'Subcluster': subclusters,
+                "communinal_storage_path": communical_path}
+        
+    except Exception as e:
+        print("Exception in _get_database_keys")
+        
+def _get_properties_keys(self, connection, db_name, schema, level=None) -> dict:
+    try:
+        properties_keys = dict()
+        if level == "schema":
+            properties_keys = _get_schema_keys(self, connection, db_name, schema)
+        if level == "database":
+            properties_keys = _get_database_keys(self, connection, db_name)
+        return properties_keys 
+    except Exception as e:
+        print("Error in finding schema keys in vertica ")
+        
+
 VerticaDialect.get_view_definition = get_view_definition
-# VerticaDialect.get_columns = get_columns
+VerticaDialect.get_columns = get_columns
 VerticaDialect._get_column_info = _get_column_info
+VerticaDialect.get_pk_constraint = get_pk_constraint
+VerticaDialect._get_extra_tags = _get_extra_tags
+VerticaDialect.get_projection_names = get_projection_names
+VerticaDialect.get_table_comment = get_table_comment
```

Comment:
these methods generally only need to be overridden if the implementation in the underlying `VerticaDialect` has issues. Could you add some comments here that explain why each of these needed to be overridden?

Reply:
we didn't want to hardcode SQL query in dialect, just the function blueprint is in dialect, used this method to dynamically overwrite functions in dialect from the vertica.py file.

Reply:
The vertica dialect should contain much more of this logic, since it is required by SQLAlchemy anyway.